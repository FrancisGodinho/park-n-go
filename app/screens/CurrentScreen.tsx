import { SafeAreaView, StyleSheet, Text, View } from "react-native";
import React, { useEffect, useState } from "react";
import { useGlobalContext } from "../utils/context";
import Headers from "../constants/Headers";
import { doc, getDoc } from "firebase/firestore";
import { db } from "../utils/Firebase";
import Colors from "../constants/Colors";
import ParkadeFinder from "../components/ParkadeFinder";

type Props = {};

const CurrentScreen = (props: Props) => {
  const { isParking, startTime, lotId } = useGlobalContext();

  const [parkingTime, setParkingTime] = useState(0);
  const [counter, setCounter] = useState<NodeJS.Timer>();
  const [lotName, setLotName] = useState();
  const [lotRate, setLotRate] = useState(0);

  useEffect(() => {
    if (lotId)
      (async () => {
        const lotSnap = await getDoc(doc(db, "lots", lotId));
        setLotName(lotSnap.data()?.name);
        setLotRate(lotSnap.data()?.rate);
      })();
  }, [lotId]);

  useEffect(() => {
    if (isParking && startTime) {
      console.log("hi", isParking, startTime);

      const interval = setInterval(() => {
        setParkingTime(
          Math.floor((new Date().getTime() - startTime.getTime()) / 1000)
        );
      }, 1000);
      setCounter(interval);
    } else if (counter) {
      console.log("yello");
      clearInterval(counter);
    }
    return () => clearInterval(counter);
  }, [isParking, startTime]);

  return (
    <SafeAreaView style={styles.container} testID="screen-view">
      {!isParking ? (
        <ParkadeFinder />
      ) : (
        <View style={{ flex: 1 }}>
          <Text style={[Headers.h1, styles.lotName]}>{lotName}</Text>
          <View style={styles.startTime}>
            <Text style={[Headers.h2, { color: Colors.darkWhite }]}>
              Started:{" "}
            </Text>
            <Text style={[Headers.h2, { color: Colors.primary }]}>
              {startTime?.toLocaleTimeString()}
            </Text>
          </View>
          <View style={styles.main}>
            <Text style={styles.time}>
              {Math.floor(parkingTime / 3600)}:
              {Math.floor((parkingTime % 3600) / 60)
                .toString()
                .padStart(2, "0")}
              :{(parkingTime % 60).toString().padStart(2, "0")}
            </Text>
            <View style={styles.money}>
              <Text style={[Headers.h2, styles.moneyText]}>
                ${lotRate * Math.floor(parkingTime / 3600)}
              </Text>
              <Text style={[Headers.h2, styles.moneyText]}>${lotRate}/hr</Text>
            </View>
          </View>
        </View>
      )}
    </SafeAreaView>
  );
};

export default CurrentScreen;

const styles = StyleSheet.create({
  container: { flex: 1, marginHorizontal: 10 },
  lotName: { textAlign: "left", marginTop: 10 },
  startTime: { flexDirection: "row" },
  main: { flex: 1, justifyContent: "center", alignItems: "center" },
  time: { fontSize: 50, fontWeight: "600", color: Colors.white },
  money: {
    width: "50%",
    flexDirection: "row",
    justifyContent: "space-between",
  },
  moneyText: { color: Colors.lightGray },
});
