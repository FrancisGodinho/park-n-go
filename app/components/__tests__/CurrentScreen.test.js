import React from "react";
import renderer from "react-test-renderer";
import { cleanup, render } from "@testing-library/react-native";
import CurrentScreen from "../../screens/CurrentScreen";

beforeAll((done) => {
  done();
});

afterAll((done) => {
  done();
});

let findTextElement = function (tree, element) {
  console.warn(tree);
  return true;
};

describe("<CurrentScreen />", () => {
  it.skip("renders correctly", () => {
    const tree = renderer.create(<CurrentScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it.skip("screen view exists", () => {
    const { getByTestId } = render(<CurrentScreen />);
    const item = getByTestId("screen-view");
    expect(item).toBeTruthy();
  });
});
