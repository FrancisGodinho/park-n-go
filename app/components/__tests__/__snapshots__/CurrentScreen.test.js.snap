// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 10,
    }
  }
  testID="screen-view"
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <Text
      style={
        Object {
          "color": "#C0C0C0",
          "fontFamily": "Avenir-Black",
          "fontSize": 25,
          "fontWeight": "900",
          "marginBottom": 10,
          "marginTop": 5,
          "textAlign": "center",
        }
      }
    >
      Find a parkade!
    </Text>
    <AIRMap
      handlePanDrag={false}
      initialRegion={null}
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      region={null}
      showsUserLocation={true}
      style={
        Object {
          "alignSelf": "stretch",
          "height": "100%",
        }
      }
    />
  </RCTSafeAreaView>
</RCTSafeAreaView>
`;
